src/lists.py

```diff
@@ -219,7 +229,9 @@ def drop_tr(x: List[T], k: int) -> List[T]:
     >>> drop_tr(x, 3)
     L(4, None)
     """
-    ...
+    return x if k==0 else drop_tr(x.tail,k-1)  # the definition of tail recursion is a little iffy to me
```

Comment:
You have tail recursion here as well. The expression

```python
return x if k==0 else drop_tr(x.tail,k-1)
```

is equivalent to

```python
if k == 0:
   return x
else:
   return drop_tr(x.tail,k-1)
```

The first branch is the base case, and you do not call anything recursive there, so that is irrelevant to whether we have tail-recursion or not.

The second branch you are just returning whatever the result of the recursion might be. You are not using anything from the call for further computations, so you don't need to remember anything in this frame in order to complete the computation, and that is the defining feature of a tail-call: the result of this function is the result of the function we call, without further processing.